responsibleai/model_monitor/components/src/compute_attribution_drift.py

```diff
@@ -0,0 +1,130 @@
+
+from sklearn.metrics import ndcg_score
+from azureml.exceptions import UserErrorException
+from feature_importance_utilities import get_model_wrapper, compute_explanations,  compute_categorical_features
+from io_utils import load_mltable_to_df, save_df_as_mltable
+
+from tabular.components.src._telemetry._loggerfactory import _LoggerFactory, track
+
+_logger = logging.getLogger(__file__)
+_ai_logger = None
+
+
+def _get_logger():
+    global _ai_logger
+    if _ai_logger is None:
+        _ai_logger = _LoggerFactory.get_logger(__file__)
+    return _ai_logger
+
+
+_get_logger()
+
+
+def parse_args():
+    # Parse arguments
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--baseline_data", type=str)
+    parser.add_argument("--production_data", type=str)
+    parser.add_argument("--feature_attribution_data", type=str)
+
+    args = parser.parse_args()
+
+    return args
+
+
+def calculate_attribution_drift(baseline_explanations, production_explanations):
+    """Compute feature attribution drift given two sets of explanations
+
+      :param baseline_explanations: list of explanations calculated using the baseline dataframe
+      :type baseline_explanations: list[float]
+      :param production_explanations: list of explanations calculated using the production dataframe
+      :type production_explanations: list[float]
+      :return: the ndcg metric between the baseline and production data
+      :rtype: float
+    """
+    true_relevance = np.asarray([baseline_explanations])
+    relevance_score = np.asarray([production_explanations])
+    feature_attribution_drift = ndcg_score(true_relevance, relevance_score)
+    # just log for now, eventually we will have to write the output
+    _logger.info("feature attribution drift calculated: {0}", feature_attribution_drift)
+    return feature_attribution_drift
+
+
+def compute_attribution_drift(task_type, target_column, baseline_dataframe, production_dataframe, feature_attribution_data):
+    """Compute feature attribution drift by calculating feature importances on each
+    dataframe input and using these to calculate the ndcg metric
+
+      :param task_type: The task type (regression or classification) of the resulting model
+      :type task_type: string
+      :param target_column: the column to predict
+      :type target_column: string
+      :param baseline_dataframe: The baseline data meaning the data used to create the
+      model monitor
+      :type baseline_dataframe: pandas.DataFrame
+      :param production_dataframe: The production data meaning the most recent set of data
+      sent to the model monitor, the current set of data
+      :type production_dataframe: pandas.DataFrame
+      :return: the ndcg metric between the baseline and production data
+      :rtype: float
+    """
+    if len(baseline_dataframe.columns.difference(production_dataframe.columns)) > 0:
+        raise UserErrorException("Dataset columns differ in baseline and production datasets")
```

Comment:
Should this be SystemError?

Reply:
imo, no because the user provides both the baseline and production data so if they don't match that's a user issue